### ISSUE TITLE
AuthError: iOS Amplify V2 Token returned by service could not be parsed

### ISSUE THREAD

**Issue by bevanchristian** [aws-amplify/amplify-swift#3107]
### Describe the bug

So, I have an application that uses Amplify v1 and has been running well for a year, and now I want to upgrade to Amplify v2.

in app delegate i configure my amplify using custom configuration

```swift
    private func initializeAmplify() {
        do {
            let authConf = AuthCategoryConfiguration(plugins: AuthConfig().buildConfig())
            let config = AmplifyConfiguration(
                        analytics: nil,
                        api: nil,
                        auth: authConf,
                        dataStore: nil,
                        hub: nil,
                        logging: nil,
                        predictions: nil,
                        storage: nil
                    )
            Amplify.Logging.logLevel = .verbose
            try Amplify.add(plugin: AWSCognitoAuthPlugin())
            try Amplify.configure(config)
        } catch {
            print("Failed to initialize Amplify with \(error)")
        }
    }

```
this is my function for build config

```swift
    func buildConfig() -> [String: JSONValue] {
            return [
                   "awsCognitoAuthPlugin": [
                       "IdentityManager": [
                           "Default": [:]
                       ],
                       "Auth": [
                           "Default": [
                                "authenticationFlowType": "USER_SRP_AUTH",
                                "OAuth": [
                                    "WebDomain": "xxxx",
                                    "AppClientId": "xxxxxx",
                                    "SignInRedirectURI": "myapp://profile/me/",
                                    "SignOutRedirectURI": "myapp://home/",
                                    "Scopes": [
                                        "aws.cognito.signin.user.admin",
                                        "email",
                                        "openid",
                                        "phone",
                                        "profile"
                                    ]
                                ]
                           ]
                       ],
                       "CognitoUserPool": [
                            "Default": [
                                "PoolId": "xxxxx",
                                "AppClientId": "xxxxxxx",
                                "Region": "ap-southeast-1"
                            ]
                       ],
                       "CredentialsProvider": [
                           "CognitoIdentity": [
                                "Default": [
                                    "PoolId": "xxxxxxxx",
                                    "Region": "ap-southeast-1"
                                ]
                           ]
                       ]
                   ]
               ]
    }
```
and i call this code for user to sign in

```swift
    let defaultPresentationAnchor = UIApplication.shared.windows.filter
        {$0.isKeyWindow}.first ?? UIWindow()
        
        do {
            let signInWeb = try await Amplify.Auth.signInWithWebUI(presentationAnchor: defaultPresentationAnchor, options: .none)
            
            if signInWeb.isSignedIn {
                return .success(())
            } else {
                return .failure(.unknown("Sign in failed or was canceled."))
            }
        } catch let error as AuthError {
            return .failure(error as? AuthError ?? .unknown("Unknown error occurred during sign-in"))
        } catch {
            return .failure(.unknown("Sign in failed or was canceled."))
        }
    }
```
    
   after user click signin its open the webview, but after that we got error 

```   
   failed AuthError: Token returned by service could not be parsed
```
   i already check in my backend the scope we define in `buildConfig` function is exactly same and its work in amplify v1
   
   my error screenshot:
![Screen Shot 2023-07-31 at 13 34 32](https://github.com/aws-amplify/amplify-swift/assets/50088801/06947802-fb67-4292-92a4-14f1a53f4d54)

full log

```shell
2023-07-31 16:56:17.563541+0700 IDN[27923:] [AWSCognitoAuthPlugin] AWSCognitoAuthPlugin/ShowHostedUISignIn.swift Sending event HostedUIEvent.fetchToken
2023-07-31 16:56:17.564436+0700 IDN[27923:709352] [AWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.configured" =     {
        "AuthenticationState.signingIn" =         {
            "SignInState.signingInWithHostedUI" =             {
                "HostedUISignInState.fetchingToken" =                 {
                };
            };
        };
        "AuthorizationState.configured" =         {
        };
    };
}
2023-07-31 16:56:17.564584+0700 IDN[27923:709352] [UserPoolSignInHelper] Checking next step for: signingInWithHostedUI(AWSCognitoAuthPlugin.HostedUISignInState.fetchingToken)
2023-07-31 16:56:17.565265+0700 IDN[27923:709351] [AWSCognitoAuthPlugin] AWSCognitoAuthPlugin/FetchHostedUISignInToken.swift Starting execution
{
    "AuthState.configured" =     {
        "AuthenticationState.signingIn" =         {
            "SignInState.signingInWithHostedUI" =             {
                "HostedUISignInState.error" =                 {
                    error = "AWSCognitoAuthPlugin.SignInError.hostedUI(AWSCognitoAuthPlugin.HostedUIError.tokenParsing)";
                };
            };
        };
        "AuthorizationState.configured" =         {
        };
    };
}
2023-07-31 16:56:18.145613+0700 IDN[27923:709351] [AWSCognitoAuthPlugin] Auth state change:

{
    "AuthState.configured" =     {
        "AuthenticationState.signedOut" =         {
            lastKnownUserName = "(nil)";
        };
        "AuthorizationState.configured" =         {
        };
    };
}
2023-07-31 16:56:18.145718+0700 IDN[27923:] [UserPoolSignInHelper] Checking next step for: signingInWithHostedUI(AWSCognitoAuthPlugin.HostedUISignInState.error(AWSCognitoAuthPlugin.SignInError.hostedUI(AWSCognitoAuthPlugin.HostedUIError.tokenParsing)))
2023-07-31 16:56:18.146382+0700 IDN[27923:] [HostedUISignInHelper] Sending cancel signIn
2023-07-31 16:56:18.146627+0700 IDN[27923:] [HostedUISignInHelper] Wait for signIn to cancel
error  in failed AuthError: Token returned by service could not be parsed
Recovery suggestion: Check if the the configuration provided are correct
```   
   

### Steps To Reproduce

```swift
Steps to reproduce the behavior:
1. init amplify with existing resource and put the amplifyconfiguration in AuthCategoryConfiguration
2. Execute Amplify.Auth.signInWithWebUI(presentationAnchor: presentationAnchor ?? defaultPresentationAnchor, options: .none)
3. it will open webview and then close the webview with error AWSCognitoAuthPlugin.HostedUIError.tokenParsing
```

### Expected behavior

signInWithWebUI work perfectly fine

### Amplify Framework Version

2.15.1

### Amplify Categories

Auth

### Dependency manager

Swift PM

### Swift version

5.7

### CLI version

not use this

### Xcode version

14.2

### Relevant log output

```shell
<details>
<summary>Log Messages</summary>


INSERT LOG MESSAGES HERE
```
</details>
```


### Is this a regression?

Yes

### Regression additional context

_No response_

### Platforms

iOS

### OS Version

13

### Device

iphone se

### Specific to simulators

_No response_

### Additional context

_No response_

**Comment by harsh62** [comment]
@bevanchristian 

There are a couple of follow up questions I wanted to ask. 
* Is this issue happening in Amplify V1?
* Are you able to login using SRP? (i.e. not using the WebUI)

**Comment by bevanchristian** [comment]
1. no its work perfectly fine in amplify v1
2. At this time I have never tried SRP, the old developer at my company immediately used WebUI

**Comment by bevanchristian** [comment]
![Screen Shot 2023-08-01 at 17 53 52](https://github.com/aws-amplify/amplify-swift/assets/50088801/23b90d81-79ea-4069-aeb8-4c7fe1789734)

current update its failed in line 37 after make request

**Comment by bevanchristian** [comment]
![Screen Shot 2023-08-01 at 18 10 04](https://github.com/aws-amplify/amplify-swift/assets/50088801/6850c774-59d1-4e03-b139-4a0e172b56c2)

when i print the decode result in line 72 here is the response

po try JSONSerialization.jsonObject(with: data, options: [])
▿ 4 elements
  ▿ 0 : 2 elements
    - key : refresh_token
    - value : xxxxxx
  ▿ 1 : 2 elements
    - key : id_token
    - value : xxxxxx xxxxxx
  ▿ 2 : 2 elements
    - key : token_type
    - value : Bearer
  ▿ 3 : 2 elements
    - key : access_token
    - value : xxxxxx

**Comment by harsh62** [comment]
@bevanchristian Thank you so much for the investigation.. I am in talks with my team to understand why the `expires_in` is not part of the response. Will revert back on the issue as soon as I have more information.

**Comment by harsh62** [comment]
@bevanchristian Would you be able to validate the fix by running `fix/expires-in-logic` branch.

**Comment by bevanchristian** [comment]
its work @harsh62 thankss for the fixing, btw when can this be merged into master

**Comment by bevanchristian** [comment]
oh I just saw the version already released in version 2.15.2 thanks for the help. I have also tried it on the master branch and it works fine. Thank you @harsh62
